src/pages/registration/new_registration/UploadFileRegistration.tsx

```diff
@@ -0,0 +1,154 @@
+import { dataTestId } from '../../../utils/dataTestIds';
+import { createUppy } from '../../../utils/uppy/uppy-config';
+import { getRegistrationWizardPath } from '../../../utils/urlPaths';
+import { DeleteIconButton } from '../../messages/components/DeleteIconButton';
+import { FileUploader } from '../files_and_license_tab/FileUploader';
+import { StartRegistrationAccordionProps } from './LinkRegistration';
+import { RegistrationAccordion } from './RegistrationAccordion';
+
+interface DeleteFileMutationParams {
+  registrationIdentifier: string;
+  fileIdentifier: string;
+}
+
+export const UploadRegistration = ({ expanded, onChange }: StartRegistrationAccordionProps) => {
+  const { t, i18n } = useTranslation();
+  const dispatch = useDispatch();
+
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const [uppy, setUppy] = useState<Uppy>();
+  const [uploadedFiles, setUploadedFiles] = useState<AssociatedFile[]>([]);
+
+  const createRegistrationMutation = useMutation({
+    mutationFn: async () => {
+      const newRegistration = await createRegistration();
+      return newRegistration.data;
+    },
+    onError: () => dispatch(setNotification({ message: t('feedback.error.create_registration'), variant: 'error' })),
+  });
+
+  const deleteFileMutation = useMutation({
+    mutationFn: async ({ registrationIdentifier, fileIdentifier }: DeleteFileMutationParams) => {
+      if (registrationIdentifier && fileIdentifier) {
+        await deleteRegistrationFile(registrationIdentifier, fileIdentifier);
+      }
+    },
+    onSuccess: () => dispatch(setNotification({ message: t('feedback.success.delete_file'), variant: 'success' })),
+    onError: () => dispatch(setNotification({ message: t('feedback.error.delete_file'), variant: 'error' })),
+  });
+
+  return (
+    <RegistrationAccordion elevation={5} expanded={expanded} onChange={onChange}>
+      <AccordionSummary
+        data-testid={dataTestId.registrationWizard.new.fileAccordion}
+        expandIcon={<ExpandMoreIcon fontSize="large" />}>
+        <CloudUploadIcon />
+        <div>
+          <Typography variant="h2">{t('registration.registration.start_with_uploading_file_title')}</Typography>
+          <Typography>{t('registration.registration.start_with_uploading_file_description')}</Typography>
+        </div>
+      </AccordionSummary>
+
+      <AccordionDetails>
+        {uppy ? (
+          <FileUploader uppy={uppy} addFile={(newFile) => setUploadedFiles((files) => [newFile, ...files])} />
+        ) : (
+          <Button
+            data-testid={dataTestId.registrationWizard.new.uploadFileButton}
+            variant="contained"
+            sx={{ alignSelf: 'center' }}
+            endIcon={<UploadIcon />}
+            loadingPosition="end"
+            loading={createRegistrationMutation.isPending}
+            onClick={() => fileInputRef.current?.click()}>
+            {t('common.select_file')}
+            <input
+              type="file"
+              hidden
```

Comment:
Har fikla mye med å få til en god løsning der vi får `<input type="file" />` i HTML, som jeg mener blir semantisk riktig. Men vil det skal se ut som en vanlig NVA-knapp og ikke som den gjør som default: 
![bilde](https://github.com/user-attachments/assets/c4adea9a-1715-4d80-ad83-8d35bf25cf8b)

Men har ikke funnet noen god måte å løse det på, derfor er det nå bare en knapp som trigger et skjult input-felt. Får ingen UU-feil i verken Axe eller ARC, men det føles feil å gjøre dette med knapp i stedet for input da. Men har brukt en del tid på det uten å komme noen vei, så foreløpig nøyer jeg meg med dette. 

Tar i mot forslag til bedre løsninger!

Reply:
Synes egentlig det virker som ei grei løsning, så lenge det ikke bryter med noen UU-regler 👍